add TiDB 5.2.0 release note (#6237)
pingcap/docs: releases/release-5.2.0.md

   User documents:

    - View the pessimistic lock-waiting events that are occurring on all TiKV nodes in the cluster: [`DATA_LOCK_WAITS`](/information-schema/information-schema-data-lock-waits.md)
    - View the deadlock errors recently occurred on a TiDB node: [`DEADLOCKS`](/information-schema/information-schema-deadlocks.md)
    - View the executing transaction on a TiDB node: [`TIDB_TRX`](/information-schema/information-schema-tidb-trx.md)

    - Optimize the user scenarios of adding indexes on tables with the `AUTO_RANDOM` or `SHARD_ROW_ID_BITS` attribute.

### Stability

- **Add TiFlash I/O traffic limit**

    This new feature is suitable for cloud storage with disk bandwidth of a small and specific size. It is disabled by default.
    
    TiFlash IO Rate Limiter provides a new mechanism to avoid excessive race for I/O resources between read and write tasks. It balances the responses to read and write tasks, and limits the rate automatically according to read/write workload.
    
    [User document](/tiflash/tiflash-configuration.md)

- **Improve stability of TiKV flow control**
 

>     TiKV introduces a new flow control mechanism to replacesupersede the previous RocksDB write stall mechanism. This mechanism controls flow at the scheduler layer, which avoids the issue of QPS drop caused by the stuck Raftstore or Apply threads when the write traffic is high.

Author: ```suggestion
    TiKV introduces a new flow control mechanism to replace the previous RocksDB write stall mechanism. Compared with the write stall mechanism, this new mechanism reduces the impact on the stability of foreground write by the following:

    When RocksDB compaction stacks, flow control is performed on the TiKV scheduler layer instead of the RocksDB layer, to avoid the following issues:
    
        - Raftstore is stuck, which is caused by RocksDB write stall.
        - Raft election times out, and the node leader is transferred as a result. 

    This new mechanism improves the flow control algorithm to mitigate QPS decrease when the write traffic is high.
```

    - View the executing transaction on a TiDB node: [`TIDB_TRX`](/information-schema/information-schema-tidb-trx.md)

    - Optimize the user scenarios of adding indexes on tables with the `AUTO_RANDOM` or `SHARD_ROW_ID_BITS` attribute.

### Stability

- **Add TiFlash I/O traffic limit**

    This new feature is suitable for cloud storage with disk bandwidth of a small and specific size. It is disabled by default.

    TiFlash IO Rate Limiter provides a new mechanism to avoid excessive race for I/O resources between read and write tasks. It balances the responses to read and write tasks, and limits the rate automatically according to read/write workload.

    [User document](/tiflash/tiflash-configuration.md)

- **Improve stability of TiKV flow control**

    TiKV introduces a new flow control mechanism to replace the previous RocksDB write stall mechanism. Compared with the write stall mechanism, this new mechanism reduces the impact on the stability of foreground write.

    In specific, when the stress of RocksDB compaction accumulates, flow control is performed at the TiKV scheduler layer instead of the RocksDB layer, to avoid the following issues:

>         - Raftstore is stuck, which is caused by RocksDB write stall.
>         - Raft election times out, and the node leader is transferred as a result.

Reviewer: ```suggestion
    - Raftstore is stuck, which is caused by RocksDB write stall.
    - Raft election times out, and the node leader is transferred as a result.
```

---
title: TiDB 5.2 Release Notes
---

# Release date: August 27, 2021

> 

Reviewer: ```suggestion

Release date: August 27, 2021

```

   User documents:

    - View the pessimistic lock-waiting events that are occurring on all TiKV nodes in the cluster: [`DATA_LOCK_WAITS`](/information-schema/information-schema-data-lock-waits.md)
    - View the deadlock errors recently occurred on a TiDB node: [`DEADLOCKS`](/information-schema/information-schema-deadlocks.md)
    - View the executing transaction on a TiDB node: [`TIDB_TRX`](/information-schema/information-schema-tidb-trx.md)

    - Optimize the user scenarios of adding indexes on tables with the `AUTO_RANDOM` or `SHARD_ROW_ID_BITS` attribute.

### Stability

- **Add TiFlash I/O traffic limit**

    This new feature is suitable for cloud storage with disk bandwidth of a small and specific size. It is disabled by default.
    
    TiFlash IO Rate Limiter provides a new mechanism to avoid excessive race for I/O resources between read and write tasks. It balances the responses to read and write tasks, and limits the rate automatically according to read/write workload.
    
    [User document](/tiflash/tiflash-configuration.md)

- **Improve stability of TiKV flow control**
 

>     TiKV introduces a new flow control mechanism to replace the previous RocksDB write stall mechanism. Compared with the write stall mechanism, this new mechanism reduces the impact on the stability of foreground write by the following:

Reviewer: ```suggestion
    TiKV introduces a new flow control mechanism to replace the previous RocksDB write stall mechanism. Compared with the write stall mechanism, this new mechanism reduces the impact on the stability of foreground write.
```

>     When RocksDB compaction stacks, flow control is performed on the TiKV scheduler layer instead of the RocksDB layer, to avoid the following issues:

Reviewer: ```suggestion
    In specific, when RocksDB compaction stacks, flow control is performed at the TiKV scheduler layer instead of the RocksDB layer, to avoid the following issues:
```

    
        - Raftstore is stuck, which is caused by RocksDB write stall.
        - Raft election times out, and the node leader is transferred as a result. 

    This new mechanism improves the flow control algorithm to mitigate QPS decrease when the write traffic is high.
 

> [User document](/tikv-configuration-file.md#storageflow-control), [#10137](https://github.com/tikv/tikv/issues/10137)

Author: ```suggestion
    [User document](/tikv-configuration-file.md#storageflow-control), [#10137](https://github.com/tikv/tikv/issues/10137)
```

    - Improve the MySQL compatibility of the built-in function`str_to_date` for the format specifiers `%b/%M/%r/%T` [#25767](https://github.com/pingcap/tidb/pull/25767)
    - Fix the issue that inconsistent binding caches might be created in multiple TiDB after recreating different bindings for the same query. [#26015](https://github.com/pingcap/tidb/pull/26015)
    - Fix the issue that the existing bindings cannot be loaded into cache after upgrade [#23295](https://github.com/pingcap/tidb/pull/23295)
    - Support ordering the result of `SHOW BINDINGS` by (`original_sql`, `update_time`) [#26139](https://github.com/pingcap/tidb/pull/26139)
    - Improve the logic of query optimization when bindings exist, and reduce optimization times of a query [#26141](https://github.com/pingcap/tidb/pull/26141)
    - Support completing the garbage collection automatically for the bindings in the "deleted" status [#26206](https://github.com/pingcap/tidb/pull/26206)
    - Support showing whether a binding is used for query optimization in the result of `EXPLAIN VERBOSE` [#26930](https://github.com/pingcap/tidb/pull/26930)

>     - Add a new status variation `last_plan_binding_update_time` status variation to view the timestamp corresponding to the binding cache in the current TiDB instance. [#26340](https://github.com/pingcap/tidb/pull/26340)

Reviewer 2: ```suggestion
    - Add a new status variation `last_plan_binding_update_time` status variation to view the timestamp corresponding to the binding cache in the current TiDB instance [#26340](https://github.com/pingcap/tidb/pull/26340)
```

>     - Support reporting an error when starting binding evolution or running `admin evolve bindings` to ban the baseline evolution (currently disabled in the on-premises TiDB version because it is an experimental feature) affecting other features. [#26333](https://github.com/pingcap/tidb/pull/26333)

Reviewer 2: ```suggestion
    - Support reporting an error when starting binding evolution or running `admin evolve bindings` to ban the baseline evolution (currently disabled in the on-premises TiDB version because it is an experimental feature) affecting other features [#26333](https://github.com/pingcap/tidb/pull/26333)
```

+ TiKV

>     - Limit memory consumption of the TiCDC sink[#10305](https://github.com/tikv/tikv/pull/10305)

Reviewer 2: ```suggestion
    - Limit memory consumption of the TiCDC sink [#10305](https://github.com/tikv/tikv/pull/10305)
```
